### PR TITLE
User Pref Editor change cleanup

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

Some small cleanup to the code in user prefs around updating the editor options.

* Adjusted user update logic to avoid mutating user object in react state directly.
* Also cleaned up a few places that needed a `useCallback()` wrapper.

Fixes WEB-39

## How to test

* Go to user preferences
* Change Editor
* Click Account in side, then click Preferences
* Ensure your Editor change is reflected
* Reload page and ensure it's still reflected.

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by gtsiolis** [comment]
Looking at this now! 👀

**Comment by geropl** [comment]
@gtsiolis Triggered again: You have to trigger "all" jobs, otherwise it won't update/install the preview correctly.

**Comment by gtsiolis** [comment]
@geropl Not sure what happened, but the preview environment is still not deployed. 🤷

**Comment by selfcontained** [comment]
I rebased main into it and w/ that latest build it deployed to https://bmh-user-pf016b26807.preview.gitpod-dev.com/workspaces ok. Hopefully it will work for you later still.

File: components/dashboard/src/user-settings/SelectIDE.tsx [hunk 1]
```diff
@@ -33,27 +32,44 @@ export default function SelectIDE(props: SelectIDEProps) {
...
-        additionalData.ideSettings = settings;
+    const actualUpdateUserIDEInfo = useCallback(
+        async (selectedIde: string, useLatestVersion: boolean) => {
+            const additionalData = user?.additionalData || {};
+            const ideSettings = additionalData.ideSettings || {};
 
-        const newUserData = await updateUser.mutateAsync({ additionalData });
-        props.updateUserContext && setUser({ ...newUserData });
-    };
+            // Avoid mutating user object in state for updates
```
Comment (svenefftinge) [inline components/dashboard/src/user-settings/SelectIDE.tsx:40]: This comment is a bit confusing without seeing the previous code. Maybe just remove it?

**Review by svenefftinge — APPROVED** [review]
LGTM

**Pull request merged**